homeassistant/components/fujitsu_hvac/config_flow.py

```diff
@@ -0,0 +1,145 @@
+"""Config flow for Fujitsu HVAC (based on Ayla IOT) integration."""
+from asyncio import timeout
+import logging
+from typing import Any
+
+from ayla_iot_unofficial import AylaAuthError, new_ayla_api
+from ayla_iot_unofficial.fujitsu_hvac import FujitsuHVAC
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import selector
+
+from .const import (
+    API_TIMEOUT,
+    AYLA_APP_ID,
+    AYLA_APP_SECRET,
+    CONF_DEVICE,
+    CONF_EUROPE,
+    DOMAIN,
+    NO_DEVICES_ERROR,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_EUROPE): bool,
+    }
+)
+
+
+async def get_devices(data: dict[str, Any]) -> dict[str, Any]:
+    """Get the list of devices from the ayla API."""
+    api = new_ayla_api(
+        data[CONF_USERNAME],
+        data[CONF_PASSWORD],
+        AYLA_APP_ID,
+        AYLA_APP_SECRET,
+        europe=data[CONF_EUROPE],
+    )
+    try:
+        async with timeout(API_TIMEOUT):
+            await api.async_sign_in()
+    except AylaAuthError as e:
+        raise InvalidAuth from e
+    except TimeoutError as e:
+        raise CannotConnect from e
+
+    devices = [
+        device
+        for device in await api.async_get_devices()
+        if isinstance(device, FujitsuHVAC)
+    ]
+    for dev in devices:
+        await dev.async_update()
+
+    return {device.device_serial_number: device for device in devices}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Fujitsu HVAC (based on Ayla IOT)."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Create empty data."""
+        self.devices: dict[str, FujitsuHVAC] = {}
+        self.credentials: dict[str, str] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                self.devices = await get_devices(user_input)
+                self.credentials = user_input
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                if len(self.devices) == 0:
+                    return self.async_abort(reason=NO_DEVICES_ERROR)
```

Comment:
only have code in the try that can fail
```suggestion
            try:
                self.devices = await get_devices(user_input)
            except CannotConnect:
                errors["base"] = "cannot_connect"
            except InvalidAuth:
                errors["base"] = "invalid_auth"
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"
            else:
                self.credentials = user_input
                if len(self.devices) == 0:
                    return self.async_abort(reason=NO_DEVICES_ERROR)
```